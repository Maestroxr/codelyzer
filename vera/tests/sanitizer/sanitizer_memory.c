#include <stdlib.h>
#include <stdio.h>

int main(int argc, char** argv) 
{
	int res = -1;
	/*
		Memory Sanitizer
	*/
	int* arr = new int[ 10 ];
	arr[ 5 ] = 0; // BOOM Uninitialized memory read   -V-
	if (arr[ 7 ]) 
	{
		printf("xx\n");
	}

	
	return res;
}