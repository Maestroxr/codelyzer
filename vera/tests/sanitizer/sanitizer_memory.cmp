C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_memory.c:6: Pointer's 'arr' malloc is missing null pointer check.
C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_memory.c:8: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_memory] Use of uninitialized value, suspect variable: 'arr', sanitizer_memory.c:8
	Used at:
		sanitizer_memory.c:8 in function 'main'
	Allocated at:
		sanitizer_memory.c:6 in function 'main'
:MemoryEnd:
