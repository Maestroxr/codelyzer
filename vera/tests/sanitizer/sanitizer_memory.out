C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_memory.c:10: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_memory]: use-of-uninitialized-value, Uninitialized value was created by a heap allocation, suspect variable identifiers: arr.
C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_memory.c:12: Memory:[C:\Work\AutoAssessment\codelyzer\vera\tests\sanitizer\sanitizer_memory]: use-of-uninitialized-value, suspect variable identifiers: arr.
